async-nats/src/header.rs

```diff
@@ -46,7 +46,7 @@ use serde::{Deserialize, Serialize};
 
 #[derive(Clone, PartialEq, Eq, Debug, Default, Deserialize, Serialize)]
 pub struct HeaderMap {
-    inner: HashMap<HeaderName, HeaderValue>,
+    inner: HashMap<HeaderName, Vec<HeaderValue>>,
```

Comment:
@caspervonb considering that we are sending 99% cases header with one value, maybe it's worth making this `VecOrString` enum to save some allocations?